RNA_folding.py

```diff
@@ -0,0 +1,329 @@
+
+    labels = {i: rna[i].upper() for i in range(len(rna))}
+    nx.draw_networkx_labels(G, pos, labels, font_size=10, font_color="whitesmoke")
+
+    nx.draw_networkx_edges(G, pos, edgelist=rna_edges, width=3.0, alpha=0.5)
+    nx.draw_networkx_edges(G, pos, edgelist=stem_edges, width=4.5, alpha=0.8, edge_color='tab:pink')
+
+    plt.savefig(fig_name + '.png')
+
+
+def build_cqm(stem_dict, min_stem, c):
+    """ Creates a Constrained Binary Model to optimize most likely stems from a dictionary of possible stems.
+
+    Args:
+        stem_dict: Dictionary with maximal stems as keys and list of weakly contained substems as values.
+
+    Returns: Constrained Binary Model.
+    """
+
+    # Create linear coefficients of -k^2, prioritizing inclusion of long stems.
+    # We depart slightly from the reference paper in this formulation.
+    linear_coeffs = {stem: -1 * (stem[1] - stem[0] + 1) ** 2 for sublist in stem_dict.values() for stem in sublist}
+
+    # Create constraints for overlapping and and substem containment.
+    quadratic_coeffs = pseudoknot_terms(stem_dict, min_stem=min_stem)
+
+    bqm = dimod.BinaryQuadraticModel(linear_coeffs, quadratic_coeffs, 'BINARY')
+
+    cqm = dimod.ConstrainedQuadraticModel()
+    cqm.set_objective(bqm)
+
+    # Add constraint disallowing overlapping substsems included in same maximal stem.
+    for stem, substems in stem_dict.items():
+        if len(substems) > 1:
+            # Add the variable for all zeros case in one-hot constraint
+            zeros = 'Null:' + str(stem)
+            cqm.add_variable(zeros, 'BINARY')
+            cqm.add_discrete(substems + [zeros], stem)
+
+    for stem1, stem2 in combinations(stem_dict.keys(), 2):
+        # Check maximal stems first.
+        if check_overlap(stem1, stem2):
+            # If maximal stems overlap, compare list of smaller stems.
+            for stem_pair in product(stem_dict[stem1], stem_dict[stem2]):
+                if check_overlap(stem_pair[0], stem_pair[1]):
+                    cqm.add_constraint(dimod.quicksum([dimod.Binary(stem) for stem in stem_pair]) <= 1)
+
+    return cqm
+
+
+def process_cqm_solution(sample_set, verbose=True):
+    """ Processes samples from solution and prints relevant information.
+
+    Prints information about the best feasible solution and returns a list of stems contained in solution.
+    Returns solution as a list of stems rather than a binary string.
+
+    Args:
+        sample_set: dimod sampleset element.
+        verbose: Boolean indicating if function should print additional information.
+
+    Returns: List of stems included in optimal solution, encoded as 4-tuples.
+    """
+
+    # Filter for feasibility
+    feasible_samples = sample_set.filter(lambda s: s.is_feasible)
+    # Check that feasible example exists
+    if not feasible_samples:
+        print('\033[93m' + '\nWarning! All solutions infeasible. You may need to try again.')
+        return None
```

Comment:
It looks like this returning `None` here leads to a later exception down the line when calling `make_plots`.  You could raise an exception here, call `sys.exit()`, or handle the `None` case in `main`.

Btw, an easy way to test the behavior is to just add `or True` to the if statement.